ouroboros-consensus-cardano/app/db-converter.hs

```diff
@@ -0,0 +1,266 @@
+        Disk.LMDBMK _ dbBackingTables <- LMDB.readWriteTransaction dbEnv (Disk.getDb (K2 "utxo"))
+        LMDB.readOnlyTransaction dbEnv $
+            LMDB.Cursor.runCursorAsTransaction'
+                LMDB.Cursor.cgetAll
+                dbBackingTables
+                (fromCodecMK $ getLedgerTables $ codecLedgerTables @(LedgerState blk))
+    pure (extLedgerSt `withLedgerTables` LedgerTables (ValuesMK values))
+
+store ::
+       ( LedgerDbSerialiseConstraints blk
+       , CanStowLedgerTables (LedgerState blk)
+       , HasLedgerTables (LedgerState blk)
+       , IsLedger (LedgerState blk)
+       )
+    => Config
+    -> SomeHasFS IO
+    -> CodecConfig blk
+    -> ExtLedgerState blk ValuesMK
+    -> IO ()
+store Config{to = Legacy, outpath} fs ccfg state =
+    writeExtLedgerState fs (encodeDiskExtLedgerState ccfg) outpath (stowLedgerTables state)
+store Config{to = Mem, outpath} fs@(SomeHasFS hasFS) ccfg state = do
+    -- write state
+    createDirectoryIfMissing hasFS True outpath
+    writeExtLedgerState fs (encodeDiskExtLedgerState ccfg) (outpath </> "state") (forgetLedgerTables state)
+    -- write tables
+    createDirectoryIfMissing hasFS True $ outpath </> "tables"
+    withFile hasFS (outpath </> "tables" </> "tvar") (WriteMode MustBeNew) $ \hf ->
+        void $
+            hPutAll hasFS hf $
+                CBOR.toLazyByteString $
+                    valuesMKEncoder (projectLedgerTables state)
+store Config{to = LMDB, outpath} fs@(SomeHasFS hasFS) ccfg state = do
+    -- write state
+    createDirectoryIfMissing hasFS True outpath
+    writeExtLedgerState fs (encodeDiskExtLedgerState ccfg) (outpath </> "state") (forgetLedgerTables state)
+    -- write tables
+    createDirectoryIfMissing hasFS True $ outpath </> "tables"
+    dbEnv <- LMDB.openEnvironment (fsToFilePath (MountPoint ".") $ outpath </> "tables") defaultLMDBLimits
+    dbState <- LMDB.readWriteTransaction dbEnv $ LMDB.getDatabase (Just "_dbstate")
+    dbBackingTables <-
+        LMDB.readWriteTransaction dbEnv $
+            lttraverse Disk.getDb (ltpure $ K2 "utxo")
+    LMDB.readWriteTransaction dbEnv $
+        Disk.withDbStateRWMaybeNull dbState $ \case
+            Nothing ->
+                ltzipWith3A Disk.initLMDBTable dbBackingTables codecLedgerTables (projectLedgerTables state)
+                    $> ((), Disk.DbState{Disk.dbsSeq = pointSlot $ getTip state})
+            Just _ -> liftIO $ throwIO $ Disk.LMDBErrInitialisingAlreadyHasState
+
+main :: IO ()
+main = withStdTerminalHandles $ do
+    cryptoInit
+    (conf, blocktype) <- getCommandLineConfig
+    case blocktype of
+        ByronBlock args   -> run conf args
+        ShelleyBlock args -> run conf args
+        CardanoBlock args -> run conf args
+  where
+    run conf args = do
+        ccfg <- configCodec . pInfoConfig <$> mkProtocolInfo args
+        let fs = SomeHasFS $ ioHasFS $ MountPoint "."
+        putStrLn "Loading..."
+        eState <- runExceptT $ load conf fs ccfg
```

Comment:
Is there a reason that the `load` code runs in `ExceptT`? It looks like the custom `Error`s in this module are fatal exceptions, so it would be fine IMO to `throwIO` them instead if we are not going to catch them.